lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -142,6 +144,62 @@ def _get_user_partition(self, user_partition_id):
 
         raise NoSuchUserPartitionError("could not find a UserPartition with ID [{}]".format(user_partition_id))
 
+    def get_parent_unit(self):
+        """
+        Finds xblock's parent unit if it exists.
+
+        Returns:
+            xblock: Returns the parent unit xblock if it exists.
+            If no parent unit exists, returns None.
+        """
+        xblock = self
+        while xblock:
+            xblock = xblock.get_parent()
```

Comment:
I was doing that to traverse the xblock's "family tree" but I think that loop reads better as:
```

while xblock:
    parent = xblock.get_parent()
    if parent is None:
        return None 
    if parent.category == "sequential":
        return xblock
    xblock = xblock.get_parent()

```